### PR TITLE
Updating performance information on export/import

### LINKED ISSUE #20703: Realm export performance heavily depends on the amount of users per file
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

import-export

### Describe the bug

I frequently need to export a realm (including users) and need to import it into another Keycloak server.

I have a realm with currently around 11500 users. Exporting the realm with users is extremely slow if --users=same_file is used. The export then exports just a few users per second and runs into a timeout after 5 minutes. After this time, only around 930 users have been exported, amounting to around 3 users per second.

If --users=different_files is used, the export is fairly fast if a low number of users per file (via the --users-per-file parameter) is used and becomes slow if a larger amount of users per file is used. With 50 users per file (the default), the whole export takes around 80 seconds (over 140 users per second), with 1000 users per file, the export takes 280 seconds (around 40 users per second), with a higher number this runs into a timeout. The amount of users per file (which I assume is also the transaction size) heavily affects the performance during the export.

The relationship between the number of users per file and the export time is not mentioned anywhere in the Keycloak documentation.

### Version

21.1.1

### Expected behavior

The number of users per file during a realm export should not impact the performance. Keycloak should internally always work with some fixed optimal transaction size, regardless of how many users it stores per file in the end. It should not matter whether I use 10 or 5000 users per file (except for some slight overhead for having more files).

I expect the export time to grow linearly with the number of users.

### Actual behavior

The realm export performance depends heavily on the amount of users per file. The worst case is --users=same_file, which is very slow. With --users=different_files, the performance gets better and better the lower the --users-per-file parameter is.

### How to Reproduce?

Create a realm with a large number of users (e.g. 10,000). Measure the time for a realm export using --users=same_file and compare this to using --users=different_files with different values for --users-per-file (50, 500, 5000).

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
Closes: #20703

This moves a note from the legacy WildFly documentation and places this in the new guides. Additional information on the CLI so people don't miss out on this information.

### CODE REVIEW

File: quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testExportHelp.unix.approved.txt [hunk 1]
```diff
@@ -131,11 +131,13 @@ Logging:
 Export:
 
 --dir <dir>          Set the path to a directory where files will be created with the exported data.
---file <file>        Set the path to a file that will be created with the exported data.
+--file <file>        Set the path to a file that will be created with the exported data. To export
+                       more than 500 users, export to a directory with different files instead.
```
Comment (pedroigor) [inline quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testExportHelp.unix.approved.txt:135]: How did you come up with the `500` number? Does it depend on the associations like roles, groups, attributes, etc?

Comment (ahus1) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testExportHelp.unix.approved.txt:135]: I've taken this from the old `importExport.adoc` documentation which I remove in this PR (it was a file which wasn't used any more).

Looking at the linked bug report, this number (and also the current default of 50) still seem to be valid.

Comment (pedroigor) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testExportHelp.unix.approved.txt:135]: My point is that the number is too generic and the explicit support for 500 and below users might be impacted by how much data you hold on your users.

I can have a hand full of users and thousands of roles, groups, attributes, etc. Which should also impact the export, right?

Perhaps, we can just say that "Depending on how many users and how much data they have, the best is to use separate files". Something along these lines.

Comment (ahus1) [reply quarkus/tests/integration/src/test/resources/org/keycloak/it/cli/dist/approvals/cli/help/HelpCommandDistTest.testExportHelp.unix.approved.txt:135]: When they have thousands of roles and groups, they will have an exponentially slowed down export anyway, and using the single files for the users won't help. I'd rather have something more actionable like "500 users", and it has been line this in the docs before.

**Review by pedroigor — APPROVED** [review]

**Review by sguilhen — APPROVED** [review]

**Pull request merged**